src/middleware/r2Middleware.ts

```diff
@@ -116,9 +116,14 @@ async function getFile(
       conditionalHeaders: parseConditionalHeaders(request.headers),
     });
   } catch (err) {
-    // Check if R2 threw a range not compatible error
-    if (err instanceof Error && err.message.includes('10039')) {
-      return new Response(undefined, { status: 416 });
+    if (err instanceof Error) {
+      if (err.message.includes('10020')) {
+        // Object name not valid, url probably has some weirdness in it
```

Comment:
Can you link docs to these errors? Maybe ref to the source of Wrangler?

Reply:
There doesn't appear to be any docs regarding these, opened cloudflare/cloudflare-docs#18098